### PR TITLE
添加两篇wechaty相关文章

### PR DESCRIPTION
**Pull request by leochen-g**
Fixes # .

Make sure these boxes are checked before creating your pull request -- thank you!

- [ ] Add Blog Header, including title, author, date.
e.g.
```
---
 layout: post
 title: "'Score Your Face Photo' a ML&Wechaty practice"
 date: 2017-09-18 09:00 +0800
 author: huyingxi
 ---
 
 > Author: [@huyingxi](https://github.com/huyingxi/wechaty_selfie) enjoying ML&Wechaty at BUPT
```
- [ ] Keep all filenames & url as lowercase, and use `-` to connect words instead of space. e.g. `2017-10-06-wechat-pc-impactor` instead of `2017-10-06-WeChat PC Impactor`,
- [ ] Embed the photo & video before publishing, save all external file to the blog `/download/2018` directory.
- [ ] Add `<!--more-->` section to just show the abstract in blog homepage.

[See more](https://github.com/Chatie/blog#contribute-guideline)

已经按照要求书写，@huan邀请

### CODE REVIEW

**Review by huan — APPROVED** [review]
Thank you very much for sharing your great blog articles to the Chatie community!

**Comment by huan** [comment]
@lijiarui Please review and help @gengchen528 to merge his PR for publishing, thank you very much!

File: _posts/2019-04-02-koa-wechaty-assistant.md [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+
+* 你是不是有闲置的微信号?
+* 你想不想有个小秘书定时提醒你将要做的事情？
+* 你是否为忘记一些纪念日而懊恼？
+* 你是否加班到很晚，而忘记了今天和别人有约？
+* 你是不是下班还记得拿快递，到家后才发现忘记了？
+* 你是不是想学习一下如何做一个微信小秘书？
+
+如果以上问题你有一条符合的话，那就可以安心读下去了，因为微信小秘书可以帮你解决大部分的问题。当然没有符合的话，也可以继续读下去，因为既然来了就说明你还是有兴趣的😆。
+
```
Comment (lijiarui) [inline _posts/2019-04-02-koa-wechaty-assistant.md]: 前言内容会被引入到博客中，读者打开博客看到两个相同的段落，所以建议删除其中一个。

File: _posts/2019-04-02-koa-wechaty-assistant.md [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+* config: koa,定时任务器，superagent的配置目录
+* mongodb: mongodb连接相关配置文件,schema设计与model的主要目录
+* untils: 抽取的共用方法
+
+### wechaty核心代码
+
+index.js
+
+微信登录，定时任务初始化，小秘书具体实现的主要文件。接口`getScheduleList`在每次登陆后会从数据库拉取未执行的定时任务并进行初始化，防止意外掉线后无法恢复定时任务。同时每次设置定时任务，接口`addSchedule`会直接向数据库中插入一条任务记录并把任务添加到定时任务列表中。每次任务执行完毕后，接口`updateSchedule`都会更新数据库中指定任务的状态，防止任务重复执行。
+
```
Comment (lijiarui) [inline _posts/2019-04-02-koa-wechaty-assistant.md:81]: 增加：```js 表示代码模块

File: _posts/2019-04-02-koa-wechaty-assistant.md [hunk 3]
```diff
@@ -0,0 +1,544 @@
...
+    bot.on('scan', onScan)
+    bot.on('login', onLogin)
+    bot.on('logout', onLogout)
+    bot.on('message', onMessage)
+    bot.on('friendship', onFriendShip)
+    bot.start()
+        .then(() => { console.log('开始登陆微信') })
+        .catch(e => console.error(e))
+    
+
```
Comment (lijiarui) [inline _posts/2019-04-02-koa-wechaty-assistant.md:229]: 增加 ``` 结束代码模块

File: _posts/2019-04-02-koa-wechaty-assistant.md [hunk 4]
```diff
@@ -0,0 +1,544 @@
...
+untils/index.js
+
+这里主要是输入关键词后的处理方法，在`index.js中`，我把用户输入的关键词根据空格来进行分词处理，放到一个数组中，然后传入`contentDistinguish（）`方法中。根据关键词的不同来进行处理是否是属于每日任务，当日定时任务，还是属于指定日期任。因为不同的定时任务类型，在时间格式上是有所区分的，每日任务我采用的是`Cron风格定时器`，类似`0 30 8 * * *`(每天8点30提醒)这种，而指定日期时间和当日任务我使用的是`new Date('2019-9-10 12:30:00')`来处理。
+同时为了兼容性，在日期处理上采用了全角替换半角的冒号格式，内容上为了更符合`秘书`的身份，将主语我全部替换成你，也处理了自己给自己创建定时任务与你给别人创建定时任务内容上的不同。
+
```
Comment (lijiarui) [inline _posts/2019-04-02-koa-wechaty-assistant.md:234]: 增加 ```js

File: _posts/2019-04-02-koa-wechaty-assistant.md [hunk 5]
```diff
@@ -0,0 +1,544 @@
...
+            scheduleObj.content = (scheduleObj.setter === scheduleObj.subscribe) ? scheduleObj.content = "亲爱的" + scheduleObj.subscribe + "，温馨提醒：" + keywordArray[3].replace('我', '你') : "亲爱的" + scheduleObj.subscribe + "，" + scheduleObj.setter + "委托我提醒你，" + keywordArray[3].replace('我', '你')
+        }
+        return scheduleObj
+    }
+    module.exports = {
+        getToday,
+        convertTime,
+        contentDistinguish
+    }
+
```
Comment (lijiarui) [inline _posts/2019-04-02-koa-wechaty-assistant.md:278]: 增加 ```

File: _posts/2019-04-02-koa-wechaty-assistant.md [hunk 6]
```diff
@@ -0,0 +1,544 @@
...
+koa就不用多说了，TJ大神继express后的又一神作，很轻量，而且摆脱了“回调地狱”的问题，放一张大神头像镇楼（发型很帅啊，哈哈）
+
+![](/download/2019/koa-wechaty-5.jpeg)
+
+koa服务默认使用3008端口，如果修改的话，需要在index.js中修改host的地址。这里目前写了三个接口，分别是添加定时任务，获取定时任务列表和更新任务列表，对应的数据库curd操作均在`mongodb/model.js`中完成
+
+config/koa.js
+
+
```
Comment (lijiarui) [inline _posts/2019-04-02-koa-wechaty-assistant.md]: 增加 ```js

File: _posts/2019-04-02-koa-wechaty-assistant.md [hunk 7]
```diff
@@ -0,0 +1,544 @@
...
+        update: (conditions) => { // 更新定时任务状态
+            return new Promise((resolve, reject) => {
+                Assistant.updateOne(conditions, { hasExpired: true }, (err, doc) => {
+                    if (err) return reject(err)
+                    return resolve(doc)
+                })
+            })
+        }
+    }
+
```
Comment (lijiarui) [inline _posts/2019-04-02-koa-wechaty-assistant.md]: 增加 ```

File: _posts/2019-04-02-koa-wechaty-assistant.md [hunk 8]
```diff
@@ -0,0 +1,544 @@
...
+    
+## 项目运行
+
+由于需要安装chromium,所以要先配置一下镜像，注意由于wechaty的限制，必须使用node10以上版本
+
+### npm或者yarn 配置淘宝源
+
+**(很重要，防止下载chromium失败，因为下载文件在150M左右，所以在执行npm run start后需要等待下载大概一两分钟以上，请耐心等待)**
+npm
+
```
Comment (lijiarui) [inline _posts/2019-04-02-koa-wechaty-assistant.md:440]: 增加 ```sh

File: _posts/2019-04-02-koa-wechaty-assistant.md [hunk 9]
```diff
@@ -0,0 +1,544 @@
...
+    npm config set registry https://registry.npm.taobao.org
+    npm config set disturl https://npm.taobao.org/dist
+    npm config set puppeteer_download_host https://npm.taobao.org/mirrors
+yarn
+
+    yarn config set registry https://registry.npm.taobao.org
+    yarn config set disturl https://npm.taobao.org/dist
+    yarn config set puppeteer_download_host https://npm.taobao.org/mirrors
+
+
```
Comment (lijiarui) [inline _posts/2019-04-02-koa-wechaty-assistant.md:455]: 增加 ```

File: _posts/2019-04-02-koa-wechaty-assistant.md [hunk 10]
```diff
@@ -0,0 +1,544 @@
...
+### 下载项目安装依赖
+
```
Comment (lijiarui) [inline _posts/2019-04-02-koa-wechaty-assistant.md:457]: 增加 ```sh

File: _posts/2019-04-02-koa-wechaty-assistant.md [hunk 11]
```diff
@@ -0,0 +1,544 @@
...
+    git clone git@github.com:gengchen528/wechat-assistant.git
+    cd wechat-assistant.git
+    npm install
+    npm run start
+
```
Comment (lijiarui) [inline _posts/2019-04-02-koa-wechaty-assistant.md]: 增加 ```

File: _posts/2020-04-02-node-wechaty-spider.md [hunk 12]
```diff
@@ -0,0 +1,374 @@
...
+* schedule: 任务调度的配置
+* superagent: 获取每日一句和天气信息
+* untils: 抽取的共用方法
+
+### 核心代码
+
+index.js
+
+关于微信的登录，定时任务的创建，发送信息的获取都在这个文件里
+
```
Comment (lijiarui) [inline _posts/2020-04-02-node-wechaty-spider.md]: 增加 ```js

File: _posts/2020-04-02-node-wechaty-spider.md [hunk 13]
```diff
@@ -0,0 +1,374 @@
...
+    bot.on('login',   onLogin)
+    bot.on('logout',  onLogout)
+    bot.on('message', onMessage)
+    bot.on('friendship', onFriendShip)
+    
+    bot.start()
+    	.then(() => console.log('开始登陆微信'))
+    	.catch(e => console.error(e))
+
+
```
Comment (lijiarui) [inline _posts/2020-04-02-node-wechaty-spider.md:200]: 增加 ```

File: _posts/2020-04-02-node-wechaty-spider.md [hunk 14]
```diff
@@ -0,0 +1,374 @@
...
+
+superagent/index.js
+
```
Comment (lijiarui) [inline _posts/2020-04-02-node-wechaty-spider.md:203]: 增加 ```js

File: _posts/2020-04-02-node-wechaty-spider.md [hunk 15]
```diff
@@ -0,0 +1,374 @@
...
+    	todayWeather:todayInfo.Day + ':' + todayInfo.WeatherText + '<br>' + '温度:' + todayInfo.Temp +  '<br>'
+    		+ todayInfo.Wind + todayInfo.WindLevel + '<br>' + '空气:' + todayInfo.PollutionLevel + '<br>'
+      }
+      return  obj
+    }
+    module.exports ={
+      getOne,getWeather
+    }
+ 
+
```
Comment (lijiarui) [inline _posts/2020-04-02-node-wechaty-spider.md:242]: 增加 ```

File: _posts/2020-04-02-node-wechaty-spider.md [hunk 16]
```diff
@@ -0,0 +1,374 @@
...
+      getOne,getWeather
+    }
+ 
+    
+## 项目运行
+
+由于需要安装chromium, 所以要先配置一下镜像，注意由于wechaty的限制，最好使用node10以上版本
+
+npm
+
```
Comment (lijiarui) [inline _posts/2020-04-02-node-wechaty-spider.md]: 增加 ```shell

File: _posts/2020-04-02-node-wechaty-spider.md [hunk 17]
```diff
@@ -0,0 +1,374 @@
...
+    npm config set registry https://registry.npm.taobao.org
+    npm config set disturl https://npm.taobao.org/dist
+    npm config set puppeteer_download_host https://npm.taobao.org/mirrors
```
Comment (lijiarui) [inline _posts/2020-04-02-node-wechaty-spider.md]: 增加 ```

File: _posts/2020-04-02-node-wechaty-spider.md [hunk 18]
```diff
@@ -0,0 +1,374 @@
...
+yarn
+
```
Comment (lijiarui) [inline _posts/2020-04-02-node-wechaty-spider.md:254]: 增加```sh

File: _posts/2020-04-02-node-wechaty-spider.md [hunk 19]
```diff
@@ -0,0 +1,374 @@
...
+    yarn config set registry https://registry.npm.taobao.org
+    yarn config set disturl https://npm.taobao.org/dist
+    yarn config set puppeteer_download_host https://npm.taobao.org/mirrors
```
Comment (lijiarui) [inline _posts/2020-04-02-node-wechaty-spider.md]: 增加 ```

File: _posts/2020-04-02-node-wechaty-spider.md [hunk 20]
```diff
@@ -0,0 +1,374 @@
...
+然后进行项目安装
+
```
Comment (lijiarui) [inline _posts/2020-04-02-node-wechaty-spider.md:261]: 增加 ```sh

File: _posts/2020-04-02-node-wechaty-spider.md [hunk 21]
```diff
@@ -0,0 +1,374 @@
...
+    git clone git@github.com:gengchen528/wechatBot.git
+    cd wechatBot
+    npm install 或 cnpm install
+
```
Comment (lijiarui) [inline _posts/2020-04-02-node-wechaty-spider.md]: 增加 ```

File: _posts/2020-04-02-node-wechaty-spider.md [hunk 22]
```diff
@@ -0,0 +1,374 @@
...
+    cd wechatBot
+    npm install 或 cnpm install
+   
+参数配置
+
+wechatBot/config/index.js
+
+ ## 项目相关配置
+    config/index.js
+
```
Comment (lijiarui) [inline _posts/2020-04-02-node-wechaty-spider.md]: 增加 ```js

File: _posts/2020-04-02-node-wechaty-spider.md [hunk 23]
```diff
@@ -0,0 +1,374 @@
...
+      AUTOREPLY:false,//自动聊天功能 默认关闭
+      AIBOTAPI:'http://www.tuling123.com/openapi/api',//图灵机器人API 注册地址http://www.turingapi.com/
+      APIKEY:'你的图灵机器人apikey',//图灵机器人apikey
+      ROOMNAME:'/^你的群名/i', //群名(请只修改中文，不要删除符号，这是正则)
+      ADDFRIENDWORD:'/你要触发的关键词/i',//自动加好友触发的关键词(请只修改中文，不要删除符号，这是正则)
+      ADDROOMWORD:'/加群/',//自动发送群图片触发关键词(请只修改中文，不要删除符号，这是正则)
+      ROOMCODEURL:'http://image.bloggeng.com/qun.png',//群二维码url链接(与本地群二维码路径选填一个)
+      ROOMLOCALPATH:'./static/qun.png',//本地群二维码图片路径（与群url选填一个）
+    }
+
```
Comment (lijiarui) [inline _posts/2020-04-02-node-wechaty-spider.md]: 增加 ```

**Review by lijiarui — CHANGES_REQUESTED** [review]
请按照我的review 进行修改，之后我会merge 他，感谢！

File: _posts/2019-04-02-koa-wechaty-assistant.md (hunk 1 above)
Comment (leochen-g) [reply _posts/2019-04-02-koa-wechaty-assistant.md]: 好的

**Comment by lijiarui** [comment]
Thanks！

**Pull request merged**